src/apidocs/ApiDocsArea.tsx

```diff
@@ -0,0 +1,230 @@
+            <AccordionPanel>
+              <DocEntryNode docs={module} heading={false} />
+            </AccordionPanel>
+          </AccordionItem>
+        ))}
+      </Accordion>
+    </>
+  );
+};
+
+interface DocEntryNodeProps extends BoxProps {
+  docs: ApiDocsEntry;
+  heading?: boolean;
+}
+
+const kindToFontSize: Record<string, any> = {
+  module: "2xl",
+  class: "lg",
+};
+
+const kindToSpacing: Record<string, any> = {
+  module: 5,
+  class: 5,
+  variable: 3,
+  function: 3,
+};
+
+const DocEntryNode = ({
+  docs: { id, kind, fullName, children, type, docString, baseClasses },
+  heading = true,
+  mt,
+  mb,
+  ...others
+}: DocEntryNodeProps) => {
+  const groupedChildren = useMemo(() => {
+    const filteredChildren = filterChildren(children);
+    return filteredChildren
+      ? groupBy(filteredChildren, (c) => c.kind)
+      : undefined;
+  }, [children]);
+
+  return (
+    <Box
+      id={fullName}
+      wordBreak="break-word"
+      mb={kindToSpacing[kind]}
+      p={kind === "variable" || kind === "function" ? 2 : undefined}
+      backgroundColor={
+        kind === "variable" || kind === "function" ? "gray.10" : undefined
+      }
+      borderRadius="md"
+      {...others}
+    >
+      <Box>
+        {heading && (
+          <>
+            <Text fontSize={kindToFontSize[kind]}>
+              <Text as="span" fontWeight="semibold">
+                {formatName(kind, fullName)}
+              </Text>
+              {nameSuffix(kind, type)}
+            </Text>
+            {baseClasses && baseClasses.length > 0 && (
+              <BaseClasses value={baseClasses} />
+            )}
+            {docString && <DocString value={docString} />}
+          </>
+        )}
+      </Box>
+      {groupedChildren && groupedChildren.size > 0 && (
+        <Box pl={kind === "class" ? 2 : 0} mt={3}>
+          <Box
+            pl={kind === "class" ? 2 : 0}
+            borderLeftWidth={kind === "class" ? 1 : undefined}
+          >
+            {["function", "variable", "class"].map(
+              (childKind) =>
+                groupedChildren?.get(childKind as any) && (
+                  <Box mb={5} key={childKind}>
+                    <Text fontWeight="lg" mb={2}>
+                      {groupHeading(kind, childKind)}
+                    </Text>
+                    {groupedChildren?.get(childKind as any)?.map((c) => (
+                      <DocEntryNode key={c.id} docs={c} />
+                    ))}
+                  </Box>
+                )
+            )}
+          </Box>
+        </Box>
+      )}
+    </Box>
+  );
+};
+
+const groupHeading = (kind: string, childKind: string): string => {
+  switch (childKind) {
+    case "variable":
+      return "Fields";
+    case "class":
+      return "Classes";
+    case "function":
+      return kind === "class" ? "Methods" : "Functions";
+    default: {
+      throw new Error("Unexpected");
+    }
+  }
+};
+
+const formatName = (kind: string, fullName: string): string => {
```

Comment:
Everything has a name field now so perhaps we can avoid removing the last part of the full name.